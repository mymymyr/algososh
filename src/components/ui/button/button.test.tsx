import renderer from 'react-test-renderer';
import { Button } from './button';
import { render, getByText, fireEvent } from '@testing-library/react';

describe('Тестирует компонент Button', () => {
    it('Кнопка с текстом', () => {
        const component = renderer.create(<Button text='test' />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Кнопка без текста', () => {
        const component = renderer.create(<Button />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Заблокированная кнопка', () => {
        const component = renderer.create(<Button disabled />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Кнопка с индикацией загрузки', () => {
        const component = renderer.create(<Button isLoader />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Проверяет корректность вызова колбека при клике на кнопку', () => {
        const mockFunction = jest.fn();

        const { container } = render(<Button onClick={mockFunction} text='test' />);

        const button = getByText(container, 'test');

        fireEvent.click(button);

        expect(mockFunction).toBeCalledTimes(1);
    })
});
