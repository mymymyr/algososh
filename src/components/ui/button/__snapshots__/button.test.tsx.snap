// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирует компонент Button Заблокированная кнопка 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирует компонент Button Кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирует компонент Button Кнопка с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирует компонент Button Кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;
